azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunner.java

```diff
@@ -905,9 +916,32 @@ private void prepareJobProperties(final ExecutableNode node) throws IOException
       }
     }
 
+    // 6. If there are any node-specific flow overrides, we also apply them
+    final Map<String, Map<String, String>> nodeParams = this.flow.getExecutionOptions()
+        .getNodeParameters();
+    if (nodeParams != null && !nodeParams.isEmpty()) {
+      applyNodeOverrides(node, nodeParams, props);
+    }
+
     node.setInputProps(props);
   }
 
+  private void applyNodeOverrides(final ExecutableNode node,
+      final Map<String, Map<String, String>> nodeParams, final Props props) {
+    if (node.getParentFlow() != null) {
+      // apply recursively top->down
+      applyNodeOverrides(node.getParentFlow(), nodeParams, props);
+    }
+    // overrides by plain node id
+    if (nodeParams.containsKey(node.getId())) {
+      props.putAll(nodeParams.get(node.getId()));
```

Comment:
Can we please use the existing `Props` hierarchy instead of applying overwrites directly on the props map? It would be something like: `overwrites.setParent(props)`

Reply:
Same for the rule 5 for flow overrides above.

Reply:
Done, this commit: https://github.com/azkaban/azkaban/pull/2597/commits/a0105146dd5c92c45ebdaf6d6187d6049645523c

This is the same way as handling of flow override props in the existing code:
https://github.com/azkaban/azkaban/blob/master/azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunner.java#L411

The thing here is that override params are not `Props` yet. So it's not as simple as calling setParent.

AFAIK the "hierarchy" is not used/needed anywhere, so this seems like adding more complexity to the code without apparent benefit? The only reason for using this style that I can see is to stick with the same style. Is that the reason?

Reply:
yes, to maintain consistency.